### PR TITLE
docs: Add example showing how to include sharp in custom lambda

### PR DESCRIPTION
**Pull request by mtliendo**
#### Description of changes:
This PR creates a new lambda example showing how to use a lambda layer with a custom NodeJsFunction

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [x] Swift
- [x] Android
- [x] Flutter
- [x] React Native

**Please add the product(s)/platform(s) affected to the PR title**
function/examples
#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ x] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/functions/examples/deploy-lambda-layer/index.mdx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  };
+}
+
+Amplify Functions are powered by [AWS Lambda](https://aws.amazon.com/lambda/), and allow you to perform a wide variety of customization through self-contained _functions_. However, there are times when a feature may not be directly supported through what Amplify exposes. Fortunately, AWS Amplify allows the use of [custom resources](https://docs.amplify.aws/react/build-a-backend/add-aws-services/custom-resources/) to be created. This enables the full flexibility of the [AWS CDK](https://docs.aws.amazon.com/cdk/api/v2/), while still providing a type-safe environment.
+
+In this section, we'll explore this feature by creating a custom Lambda function that bundles the [Sharp nodejs package](https://www.npmjs.com/package/sharp).
+
+To get started, ensure you have a Sharp layer deployed to your AWS account.
+
+## Sharp Layer Deployment
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/functions/examples/deploy-lambda-layer/index.mdx:38]: Step 1 - Deploy your Lambda layer

File: src/pages/[platform]/build-a-backend/functions/examples/deploy-lambda-layer/index.mdx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  return {
+    props: {
+      meta
+    }
+  };
+}
+
+Amplify Functions are powered by [AWS Lambda](https://aws.amazon.com/lambda/), and allow you to perform a wide variety of customization through self-contained _functions_. However, there are times when a feature may not be directly supported through what Amplify exposes. Fortunately, AWS Amplify allows the use of [custom resources](https://docs.amplify.aws/react/build-a-backend/add-aws-services/custom-resources/) to be created. This enables the full flexibility of the [AWS CDK](https://docs.aws.amazon.com/cdk/api/v2/), while still providing a type-safe environment.
+
+In this section, we'll explore this feature by creating a custom Lambda function that bundles the [Sharp nodejs package](https://www.npmjs.com/package/sharp).
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/functions/examples/deploy-lambda-layer/index.mdx:34]: cut from here and put it right below the next heading

File: src/pages/[platform]/build-a-backend/functions/examples/deploy-lambda-layer/index.mdx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    --description "Sharp layer" \
+    --license-info "Apache License 2.0" \
+    --zip-file fileb://dist/sharp-layer.zip \
+    --compatible-runtimes nodejs18.x nodejs20.x \
+    --compatible-architectures x86_64 arm64
+```
+
+Once deployed, you'll see the ARN of your Lambda layer similar to below:
+
+```sh title="example lambda layer arn"
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/functions/examples/deploy-lambda-layer/index.mdx:60]:
```suggestion
```console title="Terminal"
```

File: src/pages/[platform]/build-a-backend/functions/examples/deploy-lambda-layer/index.mdx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+arn:aws:lambda:us-east-1:211125654548:layer:sharp
+```
+
+It's *very* important to note that we'll needs to specify the version in addition to this ARN. In this example, we'll be adding this layer to a `.env.local` file with the version appended as shown:
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/functions/examples/deploy-lambda-layer/index.mdx:64]:
```suggestion
It's **very** important to specify the version in addition to this ARN. In this example, we'll be adding this layer to a `.env.local` file with the version appended as shown:
```

File: src/pages/[platform]/build-a-backend/functions/examples/deploy-lambda-layer/index.mdx [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+```sh title=".env.local"
+SHARP_LAYER_ARN=arn:aws:lambda:us-east-1:211125654548:layer:sharp:1
+```
+
+## Creating a custom Lambda Function with a Lambda Layer
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/functions/examples/deploy-lambda-layer/index.mdx:70]:
```suggestion
## Step 2 - Create a custom Lambda function with a Lambda layer
```

File: src/pages/[platform]/build-a-backend/functions/examples/deploy-lambda-layer/index.mdx [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+    runtime: Runtime.NODEJS_20_X,
+    layers: [sharpLayer],
+    bundling: {
+      externalModules: ['sharp'],
+    },
+  }
+)
+```
+
+Above you can see that we are giving our function a name, where it's handler function is located, the runtime, and that our sharp layer is to be used and considered as externally brought in as a dependency.
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/functions/examples/deploy-lambda-layer/index.mdx:121]:
```suggestion
Above you can see that we are giving our function a name – where its handler function is located – the runtime, and that our sharp layer is to be used and considered as externally brought in as a dependency.
```

File: src/pages/[platform]/build-a-backend/functions/examples/deploy-lambda-layer/index.mdx [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+
+Locally, you will likely want to test out the Sharp library. This can be done by simply installing it as a dev dependency:
+
+```sh title="installing sharp"
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/functions/examples/deploy-lambda-layer/index.mdx:125]:
```suggestion
```sh title="Terminal"
```

**Review by renebrandel — DISMISSED** [review]

**Comment by josefaidt** [comment]
ope @mtliendo did your page get deleted?

**Comment by mtliendo** [comment]
closing due to layer support being added

**Pull request closed without merging**